feat(hyprland): option to hide empty workspaces

Computerdores opened the pull request:
- `config.rs` changes are to add the `show_empty` option and fix a resulting bug where all options for all window managers had to be specified
- `hyprland.rs` changes are to implement the `show_empty` option and a small refactoring of the `get_tags` method to make the event parsing clearer (I already had to change the method anyways, so I figured I would do this while I am at it)

Computerdores committed:
feat(hyprland): option to hide empty workspaces

Computerdores committed:
fix: query workspaces when window is moved

Computerdores committed:
fix: query workspaces when moving to different monitor

Computerdores committed:
fix: use correct event type for window movement

MaxVerevkin commented:
Perhaps it is better to set `is_active: is_focused || ws.windows > 0` and utilize the existing `hide_inactive_tags` option?

Computerdores committed:
fix: set is_active instead of filtering

Computerdores commented:
> Perhaps it is better to set `is_active: is_focused || ws.windows > 0` and utilize the existing `hide_inactive_tags` option?

Good point, I've done that now. (I also updated the description of the config option to make it more clear that it is now affected by `hide_inactive_tags`)

MaxVerevkin commented on README.md:
```diff
@@ -89,6 +89,9 @@ start_hidden = false # whether the bar is initially in the 'hidden' state
 max_tag = 9 # Show only the first nine tags
 
+[wm.hyprland]
+show_empty = true # whether to consider empty workspaces as active
```
I don't understand why a new config option is needed. If someone wants to see empty workspaces, they can set `hide_inactive_tags = false`.

Computerdores replied:
The option changes whether persistent workspaces are shown when they are empty. As far as I understand, most people want them to be shown, while I do not in my current setup.

That said, I should probably adjust the options description to better reflect that.

Computerdores replied:
I have renamed the option to more acurately reflect its semantics
(sorry for the delay, I'm quite busy lately)

MaxVerevkin commented on src/wm_info_provider/hyprland.rs:
```diff
@@ -57,7 +59,7 @@ impl WmInfoProvider for HyprlandInfoProvider {
                 name: ws.name.clone(),
                 is_focused: ws.name == self.active_name,
-                is_active: true,
+                is_active: self.show_empty || ws.windows > 0,
```
I _think_ that focused workspaces, even with no windows, should be considered active and shown anyway?

Computerdores replied:
That is already the case. Whether the focused workspace is active doesn't affect whether it is shown and highlighted.

Computerdores committed:
refactor: rename new wm.hyprland.show_empty to wm.hyprland.empty_is_active

Computerdores committed:
fix: handle based on whether a workspace is persistent instead of whether it is empty

Computerdores commented:
I have reworked how the config option works and what it is called:

Since this PR makes it so on Hyprland empty workspaces are hidden in accordance with the `hide_inactive_tags` option, this meant that empty, persistent workspaces would be hidden. Since I do not believe that this behavior is what most users would expect, I originally added the config option. However, it did not address the root cause: One would expect persistent workspaces to *always* be shown. For that reason, I have reworked the config option, which is now called `always_show_persistent`, to control whether persistent workspaces will be shown even when they are empty.

With this the outstanding issues should be resolved and the PR should hopefully be ready to merge.
